packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx

```diff
@@ -577,12 +577,17 @@ export class SelectionZone extends BaseComponent<ISelectionZoneProps, {}> {
     const isAlreadySingleSelected = selection.getSelectedCount() === 1 && selection.isIndexSelected(index);
 
     if (!isAlreadySingleSelected) {
+      const isModal = selection.isModal && selection.isModal();
       selection.setChangeEvents(false);
       selection.setAllSelected(false);
       selection.setIndexSelected(index, true, true);
-      if (this.props.enterModalOnTouch && this._isTouch && selection.setModal) {
-        selection.setModal(true);
-        this._setIsTouch(false);
+      if (isModal || (this.props.enterModalOnTouch && this._isTouch)) {
```

Comment:
How is `isModal` being restored here?

Reply:
Since `isModal` is set to `true` within the body, then if it was already `true` it winds up being 'restored' to `true`.